jerry-core/jerry.c

```diff
@@ -1019,6 +1040,26 @@ jerry_create_object (void)
 } /* jerry_create_object */
 
 /**
+ * Create an empty Promise object which can be resolve/reject later
+ * by calling jerry_promise_resolve/jerry_promise_reject.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return value of the created object
+ */
+jerry_value_t
+jerry_create_promise (void)
+{
+  jerry_assert_api_available ();
+#ifndef CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
+  return ecma_op_create_promise_object (ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY), ECMA_PROMISE_EXECUTOR_EMPTY);
+#else /* CONFIG_DISABLE_ES2015_PROMISE_BUILTIN */
+  return ecma_raise_syntax_error (ECMA_ERR_MSG ("Promise profile has been disabled."));
```

Comment:
Maybe better to exclude the implementation so you get a linker error?
I don't know... returning an error adds code space.